glean-core/ios/Glean/Glean.swift

```diff
@@ -549,6 +549,15 @@ public class Glean {
         }
     }
 
+    /// Set the source tags to be applied as headers when uploading pings.
+    ///
+    /// If any of the tags is invalid nothing will be set and this function will
+    /// return `false`, although if we are not initialized yet, there won't be any validation.
```

Comment:
```suggestion
    /// return `false`.
    /// If Glean is not initialized yet, tags will not be validated at this point.
```

Separates out the 2 cases, which makes it easier to understand IMO